org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/validation/XbaseUIValidator.java

```diff
@@ -84,6 +86,19 @@ public void checkRestrictedType(XConstructorCall constructorCall) {
 		checkRestrictedType(constructorCall, XbasePackage.Literals.XCONSTRUCTOR_CALL__CONSTRUCTOR, declaringType);
 	}
 
+	@Check
+	public void checkRestrictedType(XMemberFeatureCall featureCall) {
+		if (isRestrictionCheckIgnored())
+			return;
+		if (!featureCall.isTypeLiteral())
+			return;
+		JvmIdentifiableElement feature = featureCall.getFeature();
+		if (feature instanceof JvmDeclaredType) {
+			JvmDeclaredType declaringType = (JvmDeclaredType) feature;
+			checkRestrictedType(featureCall, XbasePackage.Literals.XMEMBER_FEATURE_CALL__TYPE_LITERAL, declaringType);
```

Comment:
am not sure if we should treat nested types separately. java also gives 2 errors